R/imputation.R

```diff
@@ -0,0 +1,326 @@
+##'   of the feature standard deviations. Note that when estimating
+##'   the standard deviation of the Gaussian distribution, only the
+##'   peptides/proteins which present more than 50\% recorded values
+##'   are considered. Implemented in the `imputeLCMD::impute.MinProb`
+##'   function. See [imputeLCMD::impute.MinProb()] for details and
+##'   additional parameters.
+##'
+##' - *min*: Replaces the missing values by the smallest non-missing
+##'   value in the data.
+##'
+##' - *zero*: Replaces the missing values by 0.
+##'
+##' - *mixed*: A mixed imputation applying two methods (to be defined
+##'   by the user as `mar` for values missing at random and `mnar` for
+##'   values missing not at random, see example) on two M[C]AR/MNAR
+##'   subsets of the data (as defined by the user by a `randna`
+##'   logical, of length equal to nrow(object)).
+##'
+##' - *nbavg*: Average neighbour imputation for fractions collected
+##'   along a fractionation/separation gradient, such as sub-cellular
+##'   fractions. The method assumes that the fraction are ordered
+##'   along the gradient and is invalid otherwise.
+##'
+##'   Continuous sets `NA` value at the beginning and the end of the
+##'   quantitation vectors are set to the lowest observed value in the
+##'   data or to a user defined value passed as argument `k`. Then,
+##'   when a missing value is flanked by two non-missing neighbouring
+##'   values, it is imputed by the mean of its direct neighbours. 
+##'
+##' - *none*: No imputation is performed and the missing values are
+##'   left untouched. Implemented in case one wants to only impute
+##'   value missing at random or not at random with the *mixed*
+##'   method.
+##'
+##' The `imputeMethods()` function returns a vector with valid
+##' imputation method arguments.
+##'
+##' @references
+##'
+##' Olga Troyanskaya, Michael Cantor, Gavin Sherlock, Pat Brown,
+##' Trevor Hastie, Robert Tibshirani, David Botstein and Russ B.
+##' Altman, Missing value estimation methods for DNA microarrays
+##' Bioinformatics (2001) 17 (6): 520-525.
+##'
+##' Oba et al., A Bayesian missing value estimation method for gene
+##' expression profile data, Bioinformatics (2003) 19 (16): 2088-2096.
+##'
+##' Cosmin Lazar (2015). imputeLCMD: A collection of methods for
+##' left-censored missing data imputation. R package version
+##' 2.0. \url{http://CRAN.R-project.org/package=imputeLCMD}.
+##'
+##' Lazar C, Gatto L, Ferro M, Bruley C, Burger T. Accounting for the
+##' Multiple Natures of Missing Values in Label-Free Quantitative
+##' Proteomics Data Sets to Compare Imputation Strategies. J Proteome
+##' Res. 2016 Apr 1;15(4):1116-25. doi:
+##' 10.1021/acs.jproteome.5b00981. PubMed PMID:26906401.
+##'
+##' @rdname imputation
+##'
+##' @aliases imputeMethods impute_neighbour_average impute_knn impute_mle impute_bpca impute_mixed impute_min impute_zero impute_matrix
+##'
+##' @importFrom Rcpp sourceCpp
+##' 
+##' @useDynLib MsCoreUtils
+##'
```

Comment:
I would also add an `@author Laurent Gatto` here - so that we know who to blame if something is nor working ;)